convert2rhel/breadcrumbs.py

```diff
@@ -152,78 +159,65 @@ def _generate_json(self):
+        if not os.path.exists(rhsm_facts_path):
+            loggerinst.warning("Unable to find RHSM facts folder at '%s'.", rhsm_facts_path)
+            return
 
-def sanitize_cli_options(all_cli_options, options_to_sanitize):
-    """Change value of CLI options to asterisks to hide sensitive information.
+        data = utils.flatten(dictionary=self.data, parent_key=RHSM_CUSTOM_FACTS_NAMESPACE)
+        loggerinst.info("Writing RHSM custom facts to '%s'.", RHSM_CUSTOM_FACTS_FILE)
+        # We don't need to use `_write_obj_to_array_json` function here, because
+        # we only care about dumping the facts without having multiple copies of
+        # it.
+        _write_obj_to_array_json(path=RHSM_CUSTOM_FACTS_FILE, new_object=data)
 
-    Return back all options, but sanitized.
-    """
 
-    def sanitized_iterator():
-        elems = zip_longest(all_cli_options, all_cli_options[1:], fillvalue=None)
-
-        for (c, n) in elems:
-            # we need to handle 2 possible cases how arguments are specified
-            #  ['--argument=value']
-            #  ['--argument', 'value']
-            if "=" in c:
-                pos = c.index("=")
-
-                if c[:pos] in options_to_sanitize:
-                    yield "{0}={1}".format(c[:pos], ((len(c) - pos - 1) * "*"))
-                elif " " in c[pos:]:
-                    # if value contains a space we need to add quotes
-                    yield '{0}="{1}"'.format(c[:pos], c[pos + 1 :])
-            else:
-                if c in options_to_sanitize:
-                    yield c
-                    if n is not None:
-                        yield len(n) * "*"
-                    # we've already processed n, so we advance the iterator
-                    next(elems, None)
-                else:
-                    if " " in c:
-                        # if value contains a space we need to add quotes
-                        yield '"{0}"'.format(c)
-                    else:
-                        yield c
-
-    return " ".join(sanitized_iterator())
-
-
-def write_obj_to_array_json(path, new_object, key):
+def _write_obj_to_array_json(path, new_object={}, key=None):
     """Write new object to array defined by key in JSON file.
 
-    If the file doesn't exist, create new one and create key for inserting.
-    If the file is corrupted, append complete object (with key) as if it was new file and the
-    original content of file stays there.
+    If the file doesn't exist, create new one and create key for inserting. If
+    the file is corrupted, append complete object (with key) as if it was new
+    file and the original content of file stays there.
     """
+    default_content = {key: []} if key else new_object
+
     if not (os.path.exists(path)):
         with open(path, "a") as file:
-            file_content = {key: []}
-            json.dump(file_content, file, indent=4)
+            json.dump(default_content, file, indent=4)
 
```

Comment:
We aren't currently saving any secrets in the files that I can see but since we take the time to make the permissions restrictive, the chmod should be done here.  That way the permissions are set to 0o600 before we write anything new to the file.